guvnor-asset-mgmt/guvnor-asset-mgmt-backend/src/main/java/org/guvnor/asset/management/backend/service/AssetManagementServiceImpl.java

```diff
@@ -91,111 +90,87 @@ public void configureRepository( final String repository,
-                                final String serverURL,
-                                final Boolean deployToRuntime,
-                                final String version ) {
-        Map<String, Object> params = new HashMap<String, Object>();
-        params.put( "ProjectURI",
-                    repository );
-        params.put( "ToReleaseBranch",
-                    branch );
-        params.put( "ToReleaseVersion",
-                    version );
-        params.put( "Username",
-                    userName );
-        params.put( "Password",
-                    encodePassword( password ) );
-        params.put( "ExecServerURL",
-                    serverURL );
-        params.put( "ValidForRelease",
-                    Boolean.TRUE );
-        params.put( "DeployToRuntime",
-                    Boolean.TRUE.equals( deployToRuntime ) );
-        params.put("Owner",
-                ExecutorService.EXECUTOR_ID);
-        releaseProjectEvent.fire( new ReleaseProjectEvent( params ) );
-    }
+            POM pom = pomService.load(project.getPomXMLPath());
+            pom.getGav().setVersion(devVersion);
+            pomService.save(project.getPomXMLPath(), pom, null, "Update project version on development branch");
+        }
 
-    @Override
-    public boolean supportRuntimeDeployment() {
-        return supportRuntimeDeployment;
+        newBranchEvent.fire(new NewBranchEvent(repository, branchName, Paths.convert(branchPath), System.currentTimeMillis()) );
+
+        // create release branch
+        branchName = releaseBranch;
+        if (version != null && !version.isEmpty()) {
+            branchName = branchName + "-" + version;
+        }
+        branchPath = ioService.get(URI.create("default://" + branchName + "@" + repository));
+        branchOriginPath = ioService.get(URI.create("default://" + sourceBranch + "@" + repository));
+
+        ioService.copy(branchOriginPath, branchPath);
+
+        newBranchEvent.fire(new NewBranchEvent(repository, branchName, Paths.convert(branchPath), System.currentTimeMillis()) );
+
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put("RepositoryName", repository);
+        parameters.put("SourceBranchName", sourceBranch);
+        parameters.put("DevBranchName", devBranch);
+        parameters.put("RelBranchName", releaseBranch);
+        parameters.put("Version", version);
+        configureRepositoryEvent.fire(new ConfigureRepositoryEvent(parameters));
     }
 
+
     @Override
     public Set<Project> getProjects( final Repository repository,
                                      final String branch ) {
         return projectService.get().getProjects( repository,
                                                  branch );
     }
 
-    protected String encodePassword( final String password ) {
-        if ( password == null ) {
-            return null;
+    private Set<Project> getProjects( final Repository repository ) {
```

Comment:
that's actually as it should be as this is system operation so all projects should be updated to be consistent